apps/sync/js/sync-engine/syncengine.js

```diff
@@ -0,0 +1,274 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
```

Comment:
It would be great if you could add some comments about the overall sync process here.